mesonbuild/modules/__init__.py

```diff
@@ -83,10 +84,42 @@ def get_include_args(self, include_dirs: T.Iterable[T.Union[str, build.IncludeDi
     def find_program(self, prog: T.Union[str, T.List[str]], required: bool = True,
                      version_func: T.Optional[T.Callable[['ExternalProgram'], str]] = None,
                      wanted: T.Optional[str] = None, silent: bool = False,
-                     for_machine: MachineChoice = MachineChoice.HOST) -> 'ExternalProgram':
+                     for_machine: MachineChoice = MachineChoice.HOST) -> T.Union[ExternalProgram, OverrideProgram, 'build.Executable']:
```

Comment:
build.Executable needs to be imported in the TYPE_CHECKING block.

Reply:
That file is not checked by mypy and changing that cause a billion errors. I leave that to someone else who want to typecheck that module.

Reply:
I don't care about mypy right now, but this is easy to fix, and what it fixes is this:

```
$ flake8 mesonbuild/modules/__init__.py 
mesonbuild/modules/__init__.py:67:70: F821 undefined name 'build'
mesonbuild/modules/__init__.py:87:99: F821 undefined name 'OverrideProgram'
mesonbuild/modules/__init__.py:87:116: F821 undefined name 'build'
mesonbuild/modules/__init__.py:92:79: F821 undefined name 'OverrideProgram'
mesonbuild/modules/__init__.py:92:96: F821 undefined name 'build'
mesonbuild/modules/__init__.py:123:47: F821 undefined name 'build'
mesonbuild/modules/__init__.py:123:65: F821 undefined name 'build'
mesonbuild/modules/__init__.py:126:38: F821 undefined name 'build'
mesonbuild/modules/__init__.py:126:58: F821 undefined name 'build'
mesonbuild/modules/__init__.py:222:62: F821 undefined name 'build'
mesonbuild/modules/__init__.py:225:54: F821 undefined name 'build'
```

Not all of these are your fault :) a bunch of them come from a2def550c586aeba4269588e79a1a308467f2582.

This is what happens when flake8 doesn't pass the entire project and we don't enforce it...

Reply:
ah I see, should pass now.